docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,215 @@
+.. _vision_data_class:
+
+=====================
+The Vision Data Class
+=====================
+The :class:`VisionData <deepchecks.vision>` data class is the deepchecks base class for
+storing your data for a vision task. It is essentially a wrapper around a batch loader of images, labels,
+and predictions, that allows deepchecks to efficiently calculate different
+:doc:`checks </checks_gallery/vision>` on your data, by caching some of the information.
+
+Information about the supported task types and the required formats for each task is available
+:doc:`here </user-guide/vision/supported_tasks_and_formats>`.
+
+
+Common Class Parameters
+=======================
+
+- **batch_loader** - The batch loader can be either a pytorch DataLoader, a tensorflow Dataset or any custom
+  generator. It is important to note that the data loaded by the batch loader must be **shuffled**.
+
+- **task_type** - The task type of the data, can be either ``classification``, ``object_detection``,
+  ``semantic_segmentation``, or ``other``. Data format validation is done upon creation of VisionData based
+  on the selected task type. See :doc:`section </user-guide/vision/supported_tasks_and_formats>`
+  for more information.
+
+- **label_map** - A dictionary mapping class ids to their names.
+
+- **reshuffle_data** - Whether to reshuffle the data. Since data must be shuffled for the checks to
+  work properly, only set this to False if you are sure that the data is already shuffled.
+
+.. _vision_data__creating_vision_data:
+Creating a VisionData Object
+============================
+The sub-sections below contain simple examples how to create a VisionData object without predictions, in
+order to learn how to supply them see the :ref:`following section <vision_data__adding_predictions>`.
+
```

Comment:
1. The order of the subsections following should be different IMO (as "generic generator" should be the least common, so would put it last). More importantly, and that would solve partially also this point:
2. Please add here a short "TOC" (something like: we will explain here the three ways to create the VisionData object: 1. From Pytorch.. <links to section> 2. From Te.. <links to section> 3. Generic.. <links>)

Reply:
@Nadav-Barak

Reply:
1. I am not sure about this as generic is by far the simplest option. relevant use cases for this are also common:
if you use keras without a dataset, if you want to do a quick POC for the package for data you found in kaggle / your own data, if your data is stored on disk and you want to run validation on it before modeling.
2. will add